src/package.rs

```diff
@@ -305,20 +334,23 @@ impl Package {
 }
 
 impl TryFrom<Bytes> for Package {
-    type Error = eyre::Error;
+    type Error = miette::Report;
 
-    fn try_from(tgz: Bytes) -> eyre::Result<Self> {
+    fn try_from(tgz: Bytes) -> Result<Self, Self::Error> {
         let mut tar = Vec::new();
 
         let mut gz = flate2::read::GzDecoder::new(tgz.clone().reader());
 
         gz.read_to_end(&mut tar)
-            .wrap_err("Failed to decompress package")?;
+            .into_diagnostic()
+            .wrap_err("failed to decompress package")?;
```

Comment:
```suggestion
            .wrap_err_with(|| miette!("failed to decompress package"))?;
```

Reply:
Although i think using `.wrap_err(miette!(..))` would be more readable, it creates the error before it is actually ocuring but its much more readable and concise. I dont think we are in a memory critic context here so i would value maintainability